jupyter_collaboration/loaders.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+import asyncio
+from logging import Logger, getLogger
+from typing import Any, Callable, Coroutine
+
+from jupyter_server.utils import ensure_async
+
+
+class FileLoader:
+    def __init__(
+        self,
+        file_id: str,
+        file_format: str,
+        file_type: str,
+        file_id_manager: Any,
+        contents_manager: Any,
+        log: Logger | None,
+        poll_interval: float | None = None,
+    ) -> None:
+        self._file_id: str = file_id
+        self._file_format: str = file_format
+        self._file_type: str = file_type
+        self._last_modified = None
+
+        self._lock = asyncio.Lock()
+        self._poll_interval = poll_interval
+        self._file_id_manager = file_id_manager
+        self._contents_manager = contents_manager
+
+        self._log = log or getLogger(__name__)
+        self._subscriptions: dict[str, Callable[[str], Coroutine[Any, Any, None]]] = {}
+
+        if self._poll_interval:
+            self._watcher = asyncio.create_task(self.watch_file())
```

Comment:
Let's be sure the attribute exits

```suggestion
        self._watcher = asyncio.create_task(self.watch_file()) if self._poll_interval else None
```

Reply:
Good catch! Thanks!